src/pages/Upsolve/AtcoderUp.js

```diff
@@ -1,12 +1,97 @@
-import React from 'react'
+import React, { useEffect, useState } from 'react'
+import Validate from './Validate'
+import '../../styles/Upsolve/upsolve.css'
+import logo from '../../assets/SitesImages/Atcoder/atcoder.png'
+import refresh from '../../assets/Upsolve/reload.png'
+import { AtcoderAPI } from '../../actions/Upsolve';
+import Toggle from './Toggle';
+import Carousel_Upsolve from './CarouselUpsolve';
+import Lastpages from './Lastpages';
 
-function AtcoderUp(props) {
-    console.log(props);
+const Atcoder = () => {
+    const pageNumbers = []
+    const [page, setPage] = useState(1)
+    const [loader, setLoader] = useState(false)
+    const [prev, setPrev] = useState(null)
+    const [next, setNext] = useState(2)
+    const [first, setFirst] = useState(1)
+    const [last, setLast] = useState(null)
+    const [conData, setData] = useState([])
```

Comment:
Use understandable variables. I cant understand what `conData` means. So try to use complete words as vairable, dosen't matter if they are long.